public/app/plugins/datasource/prometheus/querybuilder/parsing.ts

```diff
@@ -350,44 +350,31 @@ function handleBinary(expr: string, node: SyntaxNode, context: Context) {
   const leftNumber = left.getChild('NumberLiteral');
   const rightNumber = right.getChild('NumberLiteral');
 
-  if (leftNumber || rightNumber) {
-    // Scalar case, just add operation.
-    if (leftNumber) {
-      // TODO: this should be already handled in case parent is binary expression as it has to be added to parent
-      //  if query starts with a number that isn't handled now.
-    } else {
-      handleExpression(expr, left, context);
-    }
-
-    if (rightNumber) {
-      // TODO: this should be already handled in case parent is binary expression as it has to be added to parent
-      //  if query starts with a number that isn't handled now.
-      visQuery.operations.push(makeBinOp(opDef, expr, right, binModifier));
-    } else {
-      handleExpression(expr, right, context);
-    }
-    return;
-  }
-
-  const leftBinary = left.getChild('BinaryExpr');
   const rightBinary = right.getChild('BinaryExpr');
 
-  if (leftBinary || rightBinary) {
-    // One of the sides is binary which means we don't really know if there is a query or just chained scalars. So
+  if (leftNumber) {
```

Comment:
This is basically the same thing as before just grouped by left/right so it's easier to read imho.